lib/widgets/profile.dart

```diff
@@ -33,6 +35,37 @@ class ProfilePage extends StatelessWidget {
       page: ProfilePage(userId: userId));
   }
 
+  /// The given user's real email address, if known, for displaying in the UI.
+  ///
+  /// Returns null if self-user isn't able to see [user]'s real email address.
+  ///
+  /// **Note:** Starting from Zulip version 7.0 (FL 163), there's a change in
+  /// the API about how self-user can access [user]'s real email address.
+  /// Search for "delivery_email" in https://zulip.com/api/register-queue.
```

Comment:
```suggestion
  /// The given user's real email address, if known, for displaying in the UI.
  ///
  /// Returns null if self-user isn't able to see [user]'s real email address.
```

The caller of this method doesn't need to know about that API change, right? It's something this method itself takes care of.

In that case it doesn't belong in a doc comment.

The reader for whom this information is relevant is one trying to understand the details of the method's implementation. That means the place this information could belong is in an implementation comment — `//` instead of `///`, and inside the method's body. I think the code already covers the fact of the API change; but the comment in the modern-server case could benefit from the 'Search for "delivery_email" …' line.